### PR TITLE
health: add CheckerFunc type

### PR DESCRIPTION
**Pull request by zombiezen**
Convenience type similar to `http.HandlerFunc`. I would like to get this into the release if possible.

### CODE REVIEW

File: health/health.go [hunk 1]
```diff
@@ -31,6 +31,11 @@ func (h *Handler) Add(c Checker) {
 	h.checkers = append(h.checkers, c)
 }
 
+// AddFunc adds a new check function to the handler.
+func (h *Handler) AddFunc(f CheckerFunc) {
```
Comment (vangent) [inline health/health.go]: Why is this necessary, won't `Add` work directly?

File: health/health.go [hunk 2]
```diff
@@ -84,3 +89,11 @@ func writeHealthy(w http.ResponseWriter) {
 type Checker interface {
 	CheckHealth() error
 }
+
+// CheckerFunc implements Checker by calling a function.
+type CheckerFunc func() error
```
Comment (vangent) [inline health/health.go]: Should it take a context?

File: health/health.go [hunk 3]
```diff
@@ -84,3 +89,11 @@ func writeHealthy(w http.ResponseWriter) {
 type Checker interface {
 	CheckHealth() error
 }
+
+// CheckerFunc implements Checker by calling a function.
```
Comment (vangent) [inline health/health.go]: I'd suggest a comment similar to `http.HandlerFunc` to make it more clear what the use case is for this type.

```
The CheckerFunc type is an adapter to allow the use of ordinary functions as Checkers. If f is a function with the appropriate signature, CheckerFunc(f) is a Checker that calls f.
```

File: health/health.go (hunk 1 above)
Comment (zombiezen) [reply health/health.go]: It's convenient for writing:

```go
healthHandler.AddFunc(func() error {
  // ...
})
```

instead of:


```go
healthHandler.Add(health.CheckerFunc(func() error {
  // ...
}))
```

([`net/http.ServeMux.HandleFunc`](https://godoc.org/net/http#ServeMux.HandleFunc) exists for similar reasons.)

It's a minor thing, so I'd be fine pulling this out.

File: health/health.go (hunk 3 above)
Comment (zombiezen) [reply health/health.go]: Done.

File: health/health.go (hunk 2 above)
Comment (zombiezen) [reply health/health.go]: The `Checker` interface intentionally does not take in a `Context` because a `Checker` should not perform anything expensive like an RPC: it should purely report on the health at a snapshot in time.

Comment (vangent) [reply health/health.go]: SGTM.

File: health/health.go (hunk 1 above)
Comment (vangent) [reply health/health.go]: I'd suggest either leaving it out, or making it clear in the docstring that this is just a shorthand. I think I'd lean toward the former, but either way is fine.
